version: '2.1'

services:
  db_node_01:
    build:
      context: mariadb
      dockerfile: Dockerfile
    container_name: db-node-01
    restart: unless-stopped
    environment:
    # TODO переменные окружения в .env
      MYSQL_DATABASE: ${DB_DATABASE}
      MYSQL_USER: ${DB_LOGIN}
      MYSQL_PASSWORD: ${DB_PASSWORD}
      MYSQL_ROOT_PASSWORD: '1'
    command: >
      --sql-mode='STRICT_TRANS_TABLES,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION'
      --general-log=ON
      --log-queries-not-using-indexes=ON
    networks:
      - arch-network
    ports:
      - ${DB_PORT}:3306
    healthcheck:
      test: "/usr/bin/mysql --user=stud --password=stud --execute \"SHOW DATABASES;\""
      timeout: 20s
      retries: 5
    volumes:
      - db-node-01-data:/var/lib/mysql

volumes:
  db-node-01-data:

networks:
  arch-network:
    driver: "bridge"
    external: false